src/Resources/RoleResource.php

```diff
@@ -46,198 +45,178 @@ protected static function getNavigationLabel(): string
-                                        $set('select_all', true); // if all toggles on => turn select_all on
-                                    }
-
-                                    if (in_array(true,$entityStates, true) === false) {
-                                        $set('select_all', false); // if even one toggle off => turn select_all off
-                                    }
-                                })
-                                ->reactive()
-                                ->afterStateUpdated(function (Closure $set, Closure $get, $state) use($entity){
-
-                                    $permissionStates = [];
-                                    foreach(static::getPermissions() as $perm) {
-                                        $permissionStates [] = $get($entity.'_'.$perm);
-                                    }
-
-                                    if (in_array(false,$permissionStates, true) === false) {
-                                        $set($entity, true); // if all permissions true => turn toggle on
-                                    }
-
-                                    if (in_array(true,$permissionStates, true) === false) {
-                                        $set($entity, false); // if even one false => turn toggle off
-                                    }
-
-                                    if(!$state) {
-                                        $set($entity,false);
-                                        $set('select_all',false);
-                                    }
-
-                                    $entityStates = [];
-                                    foreach(static::getEntities() as $ent) {
-                                        $entityStates [] = $get($ent);
-                                    }
-
-                                    if (in_array(false,$entityStates, true) === false) {
-                                        $set('select_all', true); // if all toggles on => turn select_all on
-                                    }
-
-                                    if (in_array(true,$entityStates, true) === false) {
-                                        $set('select_all', false); // if even one toggle off => turn select_all off
-                                    }
-                                });
+        return collect(static::getPermissions())->reduce(function (array $permissions, string $permission) use ($entity) {
+            $permissions[] = Forms\Components\Checkbox::make($entity . '_' . $permission)
+                ->label(__($permission))
+                ->extraAttributes(['class' => 'text-primary-600'])
+                ->afterStateHydrated(function (Closure $set, Closure $get, ?Role $record) use ($entity, $permission) {
+                    if (is_null($record)) return;
+
+                    $existed = $record->hasPermissionTo($entity . '_' . $permission);
```

Comment:
There is some issue here when there is no permission data in the database. It does not work when a table's roles and permissions are no data.

Reply:
fixed